### PR TITLE
Drop lock files

### PR DESCRIPTION
**Pull request by domesticmouse**
## Pre-launch Checklist

- [ ] I read the [Effective Dart: Style] _recently_, and have followed its advice.
- [ ] I signed the [CLA].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-devrel channel on [Discord].

[Effective Dart: Style]: https://dart.dev/guides/language/effective-dart/style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,8 @@
 */**/.metadata
 
 .dart_tool/
+
+# Lockfiles
+Podfile.lock
```
Comment (RedBrogdon) [inline .gitignore:12]: I was expecting to see `pubspec.lock` added here, but the other lock files were a surprise to me.

At the risk of opening up another can of worms, are there risks to taking those out as well? The Cocoapods docs are [pretty adamant](https://guides.cocoapods.org/using/using-cocoapods.html#should-i-check-the-pods-directory-into-source-control) about always including theirs.

Comment (domesticmouse) [reply .gitignore:12]: The `Podfile.lock` is generated by the flutter tooling at compile time, and depends on the dependency versions in `pubspec.lock`. 

You've linked the reasoning about checking in the Cocoapod dependencies, which we don't do.

Comment (craiglabenz) [reply .gitignore:12]: Good question, @RedBrogdon. I had no opinion about this before reading the docs and @domesticmouse's response, but after doing so, I think I agree with Brett. I think Cocoapods' docs are written with two big assumptions that are both untrue in our use case:

1. They live at the root of the project. In our case, the equivalent is `pubspec.yaml`, with all Cocoapods things being generated - as Brett said.
2. The project in question is a real production app with regular, distributed development. In our case, these are mostly idle samples.

**Review by craiglabenz — APPROVED** [review]
LGTM

**Comment by derKosi** [comment]
Lab missing instruction

https://codelabs.developers.google.com/codelabs/flutter-github-client?hl=en#5

Does not have a instructions to extend or change the interface nor channel or test dart files.

Thus beginners could get frustrated.

(Also I am not sure about the window to front platform specific code, but i have not tested on all platforms, maybe it is just right)

**Pull request merged**